widget/widget.go

```diff
@@ -160,23 +158,15 @@ func (w *BaseWidget) setFieldsAndRefresh(f func()) {
 
 	impl := w.getImpl()
 	if impl == nil {
-		w.Refresh()
-	} else {
-		impl.Refresh()
+		return
 	}
+	impl.Refresh()
 }
 
 // super will return the actual object that this represents.
 // If extended then this is the extending widget, otherwise it is self.
 func (w *BaseWidget) super() fyne.Widget {
-	impl := w.getImpl()
-
-	if impl == nil {
```

Comment:
Removing this nil check breaks the method signature. The nil return from `getImpl` is ignored.
This means "otherwise it is self" is no longer true.

Reply:
> This means "otherwise it is self" is no longer true.

You are right, but I think, that part of the description should be removed instead. The reason is that `BaseWidget` is not a widget (it does not have CreateRenderer method), so it would not be able to return itself as it will always panics with interface conversion error.

Reply:
Hmm, interesting. This is another case of "I thought it worked" but looking at it now I am less sure.

Reply:
I see, so should I mark this as resolved?

Reply:
Also with this change `super()` and `getImpl()` are exactly the same, so I think we should remove one of them, should I remove `getImpl()`?

Reply:
@andydotxyz any thoughts about this?

Reply:
I would lose `getImpl` yes, as that was added more recently.

Reply:
done in 31dcbd5